core/gather/driver/execution-context.js

```diff
@@ -124,14 +124,25 @@ class ExecutionContext {
 
     this._session.setNextProtocolTimeout(timeout);
     const response = await this._session.sendCommand('Runtime.evaluate', evaluationParams);
-    if (response.exceptionDetails) {
-      // An error occurred before we could even create a Promise, should be *very* rare.
-      // Also occurs when the expression is not valid JavaScript.
-      const errorMessage = response.exceptionDetails.exception ?
-        response.exceptionDetails.exception.description :
-        response.exceptionDetails.text;
-      return Promise.reject(new Error(`Evaluation exception: ${errorMessage}`));
+
+    // An error occurred before we could even create a Promise, should be *very* rare.
+    // Also occurs when the expression is not valid JavaScript.
+    const ex = response.exceptionDetails;
+    if (ex) {
+      const message = ex.exception?.description || ex.text;
+      const evaluationError = new Error(`Runtime.evaluate exception: ${message}`);
+      if (ex.exception?.description && ex.stackTrace) {
+        // The description contains the stack trace formatted as expected, if present.
+        evaluationError.stack = ex.exception.description;
```

Comment:
Could we combine the error stacks so the error stack also tells us where in LH code it occured?

Ex: https://github.com/GoogleChrome/lighthouse/blob/main/core/scripts/pptr-run-devtools.js#L87

Reply:
If we want that, we should just recurse the `error.cause` chain at join all the stacks for `errorStack`.

Reply:
I decided that modifying `stack` is probably a bad idea. Just added the extra details to the error message instead, and then was pleasantly surprised to see that v8 will parse the message for additional stack frames and add it to `stacks` itself!


Here's what printing `evaluationError` looks like now:
```
Runtime error encountered: Runtime.evaluate exception
Expression: (() => { return (function computeBenchmarkIndex() { /** * The GC-heavy benchmark that creates a stri
---- (elided)
ReferenceError: Prossmise is not defined
    at wrapInNativePromise (_lighthouse-eval.js:8:9)
    at _lighthouse-eval.js:83:8
Error: Runtime.evaluate exception
Expression: (() => { return (function computeBenchmarkIndex() { /** * The GC-heavy benchmark that creates a stri
---- (elided)
ReferenceError: Prossmise is not defined
    at wrapInNativePromise (_lighthouse-eval.js:8:9)
    at _lighthouse-eval.js:83:8
    at ExecutionContext._evaluateInContext (file:///Users/cjamcl/src/lighthouse/core/gather/driver/execution-context.js:140:31)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async ExecutionContext.evaluateAsync (file:///Users/cjamcl/src/lighthouse/core/gather/driver/execution-context.js:172:14)
    at async getBenchmarkIndex (file:///Users/cjamcl/src/lighthouse/core/gather/driver/environment.js:54:20)
    at async getBaseArtifacts (file:///Users/cjamcl/src/lighthouse/core/gather/base-artifacts.js:21:26)
    at async _setup (file:///Users/cjamcl/src/lighthouse/core/gather/navigation-runner.js:56:25)
    at async file:///Users/cjamcl/src/lighthouse/core/gather/navigation-runner.js:336:31
    at async Function.gather (file:///Users/cjamcl/src/lighthouse/core/runner.js:172:21)
    at async navigationGather (file:///Users/cjamcl/src/lighthouse/core/gather/navigation-runner.js:318:21)
    at async navigation (file:///Users/cjamcl/src/lighthouse/core/index.js:94:24)
```

Reply:
I decided that modifying `stack` is probably a bad idea. Just added the extra details to the error message instead, and then was pleasantly surprised to see that v8 will parse the message for additional stack frames and add it to `stacks` itself!


Here's what printing `evaluationError` looks like now:
```
Error: Runtime.evaluate exception
Expression: (() => { return (function computeBenchmarkIndex() { /** * The GC-heavy benchmark that creates a stri
---- (elided)
ReferenceError: Prommmise is not defined
    at wrapInNativePromise (_lighthouse-eval.js:8:9)
    at _lighthouse-eval.js:83:8
    at ExecutionContext._evaluateInContext (file:///Users/cjamcl/src/lighthouse/core/gather/driver/execution-context.js:139:31)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async ExecutionContext.evaluateAsync (file:///Users/cjamcl/src/lighthouse/core/gather/driver/execution-context.js:173:14)
    at async getBenchmarkIndex (file:///Users/cjamcl/src/lighthouse/core/gather/driver/environment.js:54:20)
    at async getBaseArtifacts (file:///Users/cjamcl/src/lighthouse/core/gather/base-artifacts.js:21:26)
    at async _setup (file:///Users/cjamcl/src/lighthouse/core/gather/navigation-runner.js:56:25)
    at async file:///Users/cjamcl/src/lighthouse/core/gather/navigation-runner.js:336:31
    at async Function.gather (file:///Users/cjamcl/src/lighthouse/core/runner.js:172:21)
    at async navigationGather (file:///Users/cjamcl/src/lighthouse/core/gather/navigation-runner.js:318:21)
    at async navigation (file:///Users/cjamcl/src/lighthouse/core/index.js:94:24)
```

Reply:
did something change? Here's the error stack I get on `hreflang` if I make an error in `LinkElements`:
```js
errorMessage: Required LinkElements gatherer encountered an error: getElementsInDocumentz is not defined
errorStack: Error: LighthouseError: ERRORED_REQUIRED_ARTIFACT
    at Runner._runAudit (~/lighthouse/core/runner.js:431:25)
    at Runner._runAudits (~/lighthouse/core/runner.js:370:40)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Runner.audit (~/lighthouse/core/runner.js:62:32)
    at async runLighthouse (~/lighthouse/cli/run.js:250:8)
    at async ~/lighthouse/cli/index.js:10:1
    at <anonymous>:1:1
```